node/src/indexer/types.rs

```diff
@@ -128,6 +136,14 @@ impl ChainSendTransactionRequest {
             ChainSendTransactionRequest::VoteReshared(_) => "vote_reshared",
         }
     }
+
+    pub fn gas_required(&self) -> Gas {
```

Comment:
Here we introduce the ability to set the attached gas based on the kind of transaction. The gas for `respond` should definitely be lowered. However, I ran into issues in the localnet tests where respond is burning ~238 Tgas for some reason. I am not sure if this is a regression after the recent contract changes or an issue in my local testing. As the gas cost is not the main focus of this PR I propose adjusting it separately after understanding the localnet issue.